js/framework/AnalyticsConstants.js

```diff
@@ -9,37 +9,356 @@ if (!OO.Analytics.EVENTS)
+     * @public
+     * @event Analytics.EVENTS#VIDEO_SEEK_REQUESTED
+     * @description This message is sent when a video seek is requested.
+     * @param {Array} paramArray Array of length 1, contains an instance of
+     * OO.Analytics.EVENT_DATA.VideoSeekStartedData.
+     */
+    VIDEO_SEEK_REQUESTED:             'video_seek_requested',
+
+    /**
+     * @public
+     * @event Analytics.EVENTS#VIDEO_SEEK_COMPLETED
+     * @description This message is sent when a video seek has completed.
+     * @param {Array} paramArray Array of length 1, contains an instance of
+     * OO.Analytics.EVENT_DATA.VideoSeekEndedData.
+     */
+    VIDEO_SEEK_COMPLETED:               'video_seek_completed',
+
+    /**
+     * @public
+     * @event Analytics.EVENTS#VIDEO_STREAM_DOWNLOADING
+     * @description This message is sent when a video stream is downloading data.
+     * If the stream has to stop because of a buffer underrun, that is considered
+     * a buffering event.
+     * @param {Array} paramArray Array of length 1, contains an instance of
+     * OO.Analytics.EVENT_DATA.VideoDownloadingMetadata.
+     */
     VIDEO_STREAM_DOWNLOADING:       'video_stream_downloading',
+
+    /**
+     * @public
+     * @event Analytics.EVENTS#VIDEO_BUFFERING_STARTED
+     * @description This message is sent when a video stream has to pause playback
+     * to load more data. It is also send when the stream is buffering before
+     * initial playback is started.
+     * @param {Array} paramArray Array of length 1, contains an instance of
+     * OO.Analytics.EVENT_DATA.VideoBufferingStartedData.
+     */
     VIDEO_BUFFERING_STARTED:        'video_buffering_started',
+
+    /**
+     * @public
+     * @event Analytics.EVENTS#VIDEO_BUFFERING_ENDED
+     * @description This message is sent when a video stream has buffered and
+     * is ready to resume playback.
+     * @param {Array} paramArray Array of length 1, contains an instance of
+     * OO.Analytics.EVENT_DATA.VideoBufferingEndedData.
+     */
     VIDEO_BUFFERING_ENDED:          'video_buffering_ended',
 
+    /**
+     * @public
+     * @event Analytics.EVENTS#AD_BREAK_STARTED
+     * @description This message is sent when the player stops the main content
+     * to start playing ads.
+     */
     AD_BREAK_STARTED:               'ad_break_started',
+
+    /**
+     * @public
+     * @event Analytics.EVENTS#AD_BREAK_ENDED
+     * @description This message is sent when the player has finished playing ads
+     * and is ready to playback the main video.
+     */
     AD_BREAK_ENDED:                 'ad_break_ended',
 
+    /**
+     * @public
+     * @event Analytics.EVENTS#DESTROY
+     * @description This message is sent when the player its plugins are destroying.
+     */
     DESTROY:                        'destroy'
   };
   OO.Analytics.EVENTS = EVENTS;
 }
 
+if (!OO.Analytics.EVENT_DATA)
+{
+  const EVENT_DATA = {};
+
+  /**
+   * @public
+   * @class Analytics.EVENT_DATA#VideoSourceData
```

Comment:
I needed to create objects with specific member variables. I don't know what else I can call them. Ideas?